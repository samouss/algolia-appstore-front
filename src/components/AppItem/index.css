.AppItem {
  display: flex;
  padding: 15px;
  border-bottom: 1px solid #f2f2f2;
  align-items: center;

  &:last-child {
    border-bottom: none;
  }

  @media (--tablet) {
    padding: 25px;
  }
}

.AppItem__Details {
  margin-left: 10px;
  overflow: hidden;
  flex-basis: 55%;

  @media (--tablet) {
    margin-left: 20px;
    flex-basis: 75%;
  }
}

.AppItem__Name {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;

  em {
    font-weight: 600;
    color: #0080ff;
  }

  @media (--tablet) {
    font-size: 1.3rem;
  }
}

.AppItem__Ratings {
  display: flex;
  font-size: 0.8rem;
  margin-top: 2px;
  align-items: center;

  @media (--tablet) {
    font-size: 1rem;
  }
}

.AppItem__Ratings__Count {
  color: #b9b9b9;
  margin-left: 5px;
  font-size: 0.85rem;

  @media (--tablet) {
    font-size: 1rem;
  }
}

.AppItem__Actions {
  margin-left: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
}
