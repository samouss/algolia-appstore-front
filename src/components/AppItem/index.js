import React from 'react';
import PropTypes from 'prop-types';
import { escapeAndReplaceTag } from 'components/Algolia';
import Thumbnail from 'components/Thumbnail';
import Ratings from 'components/Ratings';
import Button from 'components/Button';
import './index.css';

export const HitPropTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  ratingCount: PropTypes.number.isRequired,
  price: PropTypes.string.isRequired,
  _highlightResult: PropTypes.shape({
    name: PropTypes.shape({
      value: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

const AppItem = ({
  name,
  image,
  link,
  rating,
  ratingCount,
  price,
  _highlightResult,
}) => (
  <li styleName="AppItem">
    <Thumbnail
      source={image}
      alt={name}
    />

    <div styleName="AppItem__Details">
      <p
        styleName="AppItem__Name"
        dangerouslySetInnerHTML={{
          __html: escapeAndReplaceTag(_highlightResult.name.value),
        }}
      />

      <div styleName="AppItem__Ratings">
        <Ratings
          note={rating}
          on={5}
        />

        <span styleName="AppItem__Ratings__Count">
          ({ratingCount})
        </span>
      </div>
    </div>

    <div styleName="AppItem__Actions">
      <Button
        to={link}
        target="_blank"
      >
        {price === '0 $' ? 'Get' : price}
      </Button>
    </div>
  </li>
);

AppItem.propTypes = HitPropTypes;

export default AppItem;
