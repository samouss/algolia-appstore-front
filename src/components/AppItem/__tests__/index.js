import React from 'react';
import { shallow } from 'enzyme';
import AppItem from '../index';

describe('<AppItem />', () => {
  const defaultProps = {
    name: 'Cut the Rope',
    image: 'http://image.com',
    link: 'http://link.com',
    rating: 3,
    ratingCount: 1234,
    price: '0 $',
    _highlightResult: {
      name: {
        value: '__algolia__highlight__pre__tag__Cut the__algolia__highlight__post__tag__ Rope',
      },
    },
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <AppItem
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render a paid app', () => {
    const props = {
      ...defaultProps,
      price: '0.99 $',
    };

    const component = shallow(
      <AppItem
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
