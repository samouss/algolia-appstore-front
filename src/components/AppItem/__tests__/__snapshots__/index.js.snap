// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppItem /> expect to render 1`] = `
<li
  styleName="AppItem"
>
  <Thumbnail
    alt="Cut the Rope"
    source="http://image.com"
  />
  <div
    styleName="AppItem__Details"
  >
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "<em>Cut the</em> Rope",
        }
      }
      styleName="AppItem__Name"
    />
    <div
      styleName="AppItem__Ratings"
    >
      <Stars
        color="#ff9501"
        on={5}
        value={3}
      />
      <span
        styleName="AppItem__Ratings__Count"
      >
        (
        1,234
        )
      </span>
    </div>
  </div>
  <div
    styleName="AppItem__Actions"
  >
    <Button
      target="_blank"
      to="http://link.com"
    >
      Get
    </Button>
  </div>
</li>
`;

exports[`<AppItem /> expect to render a paid app 1`] = `
<li
  styleName="AppItem"
>
  <Thumbnail
    alt="Cut the Rope"
    source="http://image.com"
  />
  <div
    styleName="AppItem__Details"
  >
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "<em>Cut the</em> Rope",
        }
      }
      styleName="AppItem__Name"
    />
    <div
      styleName="AppItem__Ratings"
    >
      <Stars
        color="#ff9501"
        on={5}
        value={3}
      />
      <span
        styleName="AppItem__Ratings__Count"
      >
        (
        1,234
        )
      </span>
    </div>
  </div>
  <div
    styleName="AppItem__Actions"
  >
    <Button
      target="_blank"
      to="http://link.com"
    >
      0.99 $
    </Button>
  </div>
</li>
`;
